source/common/ssl/context_config_impl.cc

```diff
@@ -118,6 +86,32 @@ ContextConfigImpl::ContextConfigImpl(const envoy::api::v2::auth::CommonTlsContex
   }
 }
 
+void ContextConfigImpl::readCertChainConfig(const envoy::api::v2::auth::CommonTlsContext& config) {
+  if (!config.tls_certificates().empty()) {
+    cert_chain_ = Config::DataSource::read(config.tls_certificates()[0].certificate_chain(), true);
+    private_key_ = Config::DataSource::read(config.tls_certificates()[0].private_key(), true);
+    return;
+  }
+  if (!config.tls_certificate_sds_secret_configs().empty()) {
+    auto secret_name = config.tls_certificate_sds_secret_configs()[0].name();
+    if (!config.tls_certificate_sds_secret_configs()[0].has_sds_config()) {
+      // static secret
+      const auto secret = secret_manager_.findStaticTlsCertificate(secret_name);
+      if (secret) {
+        cert_chain_ = secret->certificateChain();
+        private_key_ = secret->privateKey();
+        return;
+      } else {
+        throw EnvoyException(fmt::format("Unknown static secret: {}", secret_name));
+      }
+    } else {
+      secret_provider_ = secret_manager_.findOrCreateDynamicSecretProvider(
+          config.tls_certificate_sds_secret_configs()[0].sds_config(), secret_name);
+      return;
+    }
+  }
+}
```

Comment:
don't we need to throw some type of config error here if neither of the above cases are true?

Reply:
I think it is okay to do nothing here. Without this change, the readConfig() returns empty string. 
https://github.com/envoyproxy/envoy/blob/master/source/common/ssl/context_config_impl.cc#L19
In that case cert_chain_ and private_key_ are empty. I am trying to make this readCertChainConfig() consistent with existing behavior.